cpp/include/rapidsmpf/streaming/cudf/owning_wrapper.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+
+namespace rapidsmpf::streaming {
+
+/**
+ * @brief Utility class to store an arbitrary type-erased object while another object is
+ * alive.
+ *
+ * When sending messages through `Channel`s from Python, we typically need to keep various
+ * Python objects alive since the matching C++ objects only hold views.
+ *
+ * For example, when constructing a `TableChunk` from a pylibcudf `Table`, the
+ * `TableChunk` has a non-owning `cudf::table_view` of the `Table` and someone must be
+ * responsible for keeping the `Table` alive for the lifetime of the `TableChunk`. If we
+ * want to allow creation of such objects in Python with the ability to sink them on the
+ * C++ side, we cannot rely on the Python side of things keeping the `Table` alive (the
+ * reference disappears!). Similarly when we send a message through a `Channel` the sender
+ * will, once pushed into the channel, drop the reference to the message payload and so,
+ * again, we need some way of keeping the payload alive.
+ *
+ * To square this circle, such C++ objects have an `OwningWrapper` slot that stores a
+ * type-erased pointer with, as far as we are concerned, unique ownership semantics. When
+ * this object is destroyed, the customer deleter runs and can do whatever deallocation is
```

Comment:
```suggestion
 * this object is destroyed, the custom deleter runs and can do whatever deallocation is
```
Tiny drive-by suggestion :)